packages/@aws-cdk/aws-logs/test/test.loggroup.ts

```diff
@@ -242,3 +316,14 @@ export = {
     test.done();
   },
 };
+
+function dataDrivenTests(cases: any[][], body: (test: Test, ...args: any[]) => void) {
+  const ret: any = {};
+  for (let i = 0; i < cases.length; i++) {
+    const args = cases[i]; // Need to capture inside loop for safe use inside closure.
+    ret[`case ${i + 1}`] = function(test: Test) {
+      return body.apply(this, [test, ...args]);
+    };
+  }
+  return ret;
+}
```

Comment:
I would have much preferred a simple for loop inside the test than all this. This makes the test logic more complex, IMO.

```ts
['', ':*'].forEach((suffix) => {
  ...
});
```

Reply:
Yeah... that's true. But now it's 2 different tests, so it's a more correct picture (imo).